fstest/fstests/fstests.go

```diff
@@ -1407,6 +1407,63 @@ func Run(t *testing.T, opt *Opt) {
 
 		})
 
+		// TestPutUnknownSize ensures Fs.Put() and Object.Update() don't panic when
+		// src.Size() == -1
+		t.Run("FsPutUnknownSize", func(t *testing.T) {
+
+			testPutUnknownSize := func() {
+				unknownSizePutFile := fstest.Item{
+					ModTime: fstest.Time("2002-02-03T04:05:06.499999999Z"),
+					Path:    "unknown-size-put.txt",
+				}
+
+				defer func() {
+					assert.Nil(t, recover(), "Fs.Put() should not panic when src.Size() == -1")
+				}()
+
+				contents := fstest.RandomString(100)
+				buf := bytes.NewBufferString(contents)
+				uploadHash := hash.NewMultiHasher()
```

Comment:
You don't appear to use the uploadHash...

You could check it, however I think in this test just checking the object returned from Put has Size() 100 would be a sufficient test.

If you are doing that then you can lose the fstest.Item construction too.